rpc/src/services/dev_services.rs

```diff
@@ -162,27 +165,81 @@ pub(crate) fn get_stats_memory_protocol_runners() -> MemoryStatsResult<Vec<Memor
 pub(crate) fn get_cycle_length_for_block(
     block_hash: &BlockHash,
     env: &RpcServiceEnvironment,
-    log: &Logger,
+    _: &Logger,
 ) -> Result<i32, RpcServiceError> {
-    if let Ok(context_proto_params) = get_context_protocol_params(block_hash, env) {
-        Ok(tezos_messages::protocol::get_constants_for_rpc(
-            &context_proto_params.constants_data,
-            &context_proto_params.protocol_hash,
-        ).map_err(|e| RpcServiceError::UnexpectedError {
-            reason: format!("{}", e)
-        })?
-            .map(|constants| constants.get("blocks_per_cycle")
-                .map(|value| if let UniversalValue::Number(value) = value { *value } else {
-                    slog::warn!(log, "Cycle length missing"; "block" => block_hash.to_base58_check());
-                    4096
-                })
-            ).flatten().unwrap_or_else(|| {
-            slog::warn!(log, "Cycle length missing"; "block" => block_hash.to_base58_check());
-            4096
-        }))
+    // get the protocol hash
+    let protocol_hash =
+        match BlockMetaStorage::new(env.persistent_storage()).get_additional_data(block_hash)? {
```

Comment:
@adonagy 
you can just use here cached version: `base_services::get_additional_data_or_fail`